src/censor/BuiltinStrategies.ts

```diff
@@ -139,25 +139,32 @@ export function fixedCharCensorStrategy(char: string): TextCensorStrategy {
 
 /**
  * A text censoring strategy that generates replacement strings made up of
- * random characters from the set of characters provided.
+ * random characters from the set of characters provided. The strings never
+ * contain two of the same character in a row.
  *
  * @example
  * ```typescript
  * const strategy = randomCharFromSetCensorStrategy('$#!');
  * const censor = new TextCensor().setStrategy(strategy);
  * // Before: 'fuck you!'
- * // After: '!##$ you!'
+ * // After: '!#$# you!'
  * ```
  * @param charset - Set of characters from which the replacement string should
- * be constructed. Must not be empty.
+ * be constructed. Must have at least two characters.
  * @returns A [[TextCensorStrategy]] for use with the [[TextCensor]].
  */
 export function randomCharFromSetCensorStrategy(charset: string): TextCensorStrategy {
 	const chars = [...charset];
-	if (chars.length === 0) throw new Error('The character set passed must not be empty.');
+	if (chars.length < 2) throw new Error('The character set passed must have at least 2 characters.');
 	return (ctx: CensorContext) => {
 		let censored = '';
-		for (let i = 0; i < ctx.matchLength; i++) censored += chars[Math.floor(Math.random() * chars.length)];
+		let prev = chars.length;
+		for (let i = 0; i < ctx.matchLength; i++) {
+			let idx = Math.floor(Math.random() * (i ? chars.length - 1 : chars.length));
+			if (idx >= prev) idx++;
```

Comment:
It took me a second to realize the code is written the way it is to ensure a uniform distribution of generated characters at each iteration. I think it would be worth adding a brief comment to that effect, and wonder if separating the first iteration from the rest would help clarity:
```js
if (ctx.matchLength === 0) return ''; // at the start of the function

const firstIdx = Math.floor(Math.random() * chars.length);
let censored = chars[firstIdx];
let prev = firstIdx;
for (let i = 1; i < ctx.matchLength; i++) {
	let idx = Math.floor(Math.random() * (chars.length - 1));
	if (idx >= prev) idx++;
	prev = idx;
	censored += chars[idx];
}
return censored;
```
(To be precise on what I'm requesting, I would like you to add the comment, but separating out the first iteration is optional; I'm happy to leave it as is if you think it doesn't help structurally.)